src/omnisharp/messageType.ts

```diff
@@ -0,0 +1,178 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IObserver } from "rx";
+import { PlatformInformation } from "../platform";
+import { Request } from "./requestQueue";
+import * as protocol from './protocol';
+
+export type MessageObserver = IObserver<Message>;
+export enum MessageType {
+    ActivationFailure,
+    CommandShowOutput,
+    CommandDotNetRestoreStart,
+    CommandDotNetRestoreProgress,
+    CommandDotNetRestoreSucceeded,
+    CommandDotNetRestoreFailed,
+    DebuggerNotInstalledFailure,
+    DebuggerPreRequisiteFailure,
+    DebuggerPreRequisiteWarning,
+    DownloadEnd,
+    DownloadProgress, 
+    DownloadStart, 
+    InstallationFailure,
+    InstallationSuccess, 
+    InstallationProgress, 
+    OmnisharpDelayTrackerEventMeasures,
```

Comment:
For example, this is a really specific message type. Should anyone trying to send a message have to see this in their completion list?